src/collections/componentSet.ts

```diff
@@ -474,6 +529,24 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
     return this.destructiveChangesType;
   }
 
+  /**
+   * Will return the types of destructive changes in the component set
+   *
+   * @return DestructiveChangesType[]
+   */
+  public getTypesOfDestructiveChanges(): DestructiveChangesType[] {
```

Comment:
`hasDeletes` is doing pretty much the same logic, and is checked in both the Converter and CS's getObject.

This could also return an empty array for "there are none" so there is only one.

As written, if you were to NOT check hasDeletes before asking `getTypesOfDestructiveChanges`, your CS would tell that both types are present (the else condition here).

suggestion: make an array, and conditionally push to it (there might be 0,1, or 2 values).

then we can eliminate the `hasDeletes` since this code would have no effect for an empty array.
```
const destructiveChangesTypes = cs.getTypesOfDestructiveChanges();
              destructiveChangesTypes.map((destructiveChangesType) => {
```